firstpartydata/first_party_data.go

```diff
@@ -195,42 +199,19 @@ func resolveUser(fpdConfig *openrtb_ext.ORTB2, bidRequestUser *openrtb2.User, gl
 		newUser.Data = openRtbGlobalFPD[userDataKey]
 	}
 	if fpdConfigUser != nil {
-		if err := mergeUser(newUser, fpdConfigUser); err != nil {
+		if mergedUser, err := merge.User(newUser, fpdConfigUser); err != nil {
+			if err == merge.ErrBadOverride {
+				return nil, ErrBadFPD
+			}
 			return nil, err
+		} else {
+			newUser = mergedUser
```

Comment:
When `err != nil` evaluates to `true` we end up returning `nil` in both branches, do we really need `mergedUser` as a separate variable? We could even spare the `else` clause:
```
201     if fpdConfigUser != nil {
202 -       if mergedUser, err := merge.User(newUser, fpdConfigUser); err != nil {
    +       if newUser, err := merge.User(newUser, fpdConfigUser); err != nil {
203             if err == merge.ErrBadOverride {
204                 return nil, ErrBadFPD
205             }
206             return nil, err
207 -       } else {
208 -           newUser = mergedUser
209         }
210     }
firstpartydata/first_party_data.go
```
But I see this pattern repeated in other places below (`mergedSite` and `mergedApp`) should we modify to make it leaner? Also, why do `merge.User`, `merge.Site` and `merge.App` are now returning pointers? Is it to check for shared memory issues?

Reply:
> But I see this pattern repeated in other places below (mergedSite and mergedApp) should we modify to make it leaner? 

Yes. My original intent was to not assign mergedUser if there is an error, but if there is an error there is no return path for newUser. Updated.

> Also, why do merge.User, merge.Site and merge.App are now returning pointers?

They always did this. Before, they changed the memory address of the input pointer. Now it returns a separate pointer and leaves the input alone.